tools/ci.psm1

```diff
@@ -587,9 +593,244 @@ function Invoke-AppveyorFinish
+            # This allows us to prevent regressions to a critical engine experimental feature.
+            $sudoPesterParam.Remove('Path')
+        } 
+        else 
+        {
+            # If a non-empty string or array is specified for the feature name, we only run those test files.
+            $sudoPesterParam['Path'] = $testFiles
+        }
+        $passThruResult = Start-PSPester @sudoPesterParam -Title "Pester Experimental Sudo - $featureName"
+        $sudoResultsWithExpFeatures += $passThruResult
+    }
+
+    # Determine whether the build passed
+    try {
+        $allTestResultsWithNoExpFeature = @($pesterPassThruNoSudoObject, $pesterPassThruSudoObject)
+        $allTestResultsWithExpFeatures = $noSudoResultsWithExpFeatures + $sudoResultsWithExpFeatures
+        # This throws if there was an error:
+        $allTestResultsWithNoExpFeature | ForEach-Object { Test-PSPesterResults -ResultObject $_ }
+        $allTestResultsWithExpFeatures  | ForEach-Object { Test-PSPesterResults -ResultObject $_ -CanHaveNoResult }
+        $result = "PASS"
+    } catch {
+        # The build failed, set the result:
+        $resultError = $_
+        $result = "FAIL"
+    }
+
+    try {
+        $ParallelXUnitTestResultsFile = "$pwd/ParallelXUnitTestResults.xml"
+        Start-PSxUnit -ParallelTestResultsFile $ParallelXUnitTestResultsFile
+        # If there are failures, Test-XUnitTestResults throws
+        Test-XUnitTestResults -TestResultsFile $ParallelXUnitTestResultsFile
+    } catch {
+        $result = "FAIL"
+        if (!$resultError)
+        {
+            $resultError = $_
+        }
+    }
+
+    if ($createPackages)
+    {
+        $packageParams = @{}
+        $packageParams += @{ReleaseTag=$releaseTag}
+
+        # Only build packages for PowerShell/PowerShell repository
+        # branches, not pull requests
+        $packages = @(Start-PSPackage @packageParams -SkipReleaseChecks)
+        foreach($package in $packages)
+        {
+            if (Test-Path $package)
+            {
+                Write-Log "Package found: $package"
+            }
+            else
+            {
+                Write-Error -Message "Package NOT found: $package"
+            }
+            # Publish the packages to the nuget feed if:
+            # 1 - It's a Daily build (already checked, for not a PR)
+            # 2 - We have the info to publish (NUGET_KEY and NUGET_URL)
+            # 3 - it's a nupkg file
+            if($isDailyBuild -and $NugetKey -and $env:NUGET_URL -and [system.io.path]::GetExtension($package) -ieq '.nupkg')
+            {
+                if (Test-Path $package)
+                {
+                    Write-Log "Package found: $package"
+                }
+                else
+                {
+                    Write-Error -Message "Package NOT found: $package"
+                }
+                Write-Log "pushing $package to $env:NUGET_URL"
```

Comment:
Add a newline after the code block end and new code starts.

This make the code more readable for most people.

Reply:
fixed